beetsplug/fetchart.py

```diff
@@ -216,9 +218,17 @@ def _logged_get(log, *args, **kwargs):
     else:
         message = 'getting URL'
 
+    retry_strategy = Retry(
+        total=3,
+        backoff_factor=1,
+        status_forcelist=[429, 500, 502, 503, 504, 403],
+        method_whitelist=["GET"]
```

Comment:
It sounds like the requests version should be pinned then - probably in a separate PR. 

The latest version of requests (2.31.0) still uses urllib3 version [1.2x](https://github.com/psf/requests/blob/main/setup.cfg#L10), although it looks like they are preparing for version 2.0.0. The options would the be to pin the current requests version and then merge this PR, or wait until requests have fully updated to urllib3 2.0.0, pin that version, and update this PR. Which feels preferable?

Reply:
> The latest version of requests (2.31.0) still uses urllib3 version [1.2x](https://github.com/psf/requests/blob/main/setup.cfg#L10)

I think that's for some kind of test environment, [their `setup.py` goes up to v3
](https://github.com/psf/requests/blob/6e5b15d542a4e85945fd72066bb6cecbc3a82191/setup.py#L64)
A regular `pip install requests` gave me v2.0.2 of `urllib3`.





Just checked & it looks like we don't _have_ to pin `requests`, we can pin `urllib3>=2.0.0` directly instead ([here](https://github.com/beetbox/beets/blob/a19bc0570cd462aa8c978634fa82b1049330fcd1/setup.py#L138)) 
Remains to be seen if this would affect other plugins though...

Reply:
I wonder if we could do something clever here… do we actually need the `allowed_methods`? It seems like we're probably only making GET requests here anyway, so it could be safe to just leave this off.

Reply:
True. Maybe that’s the simplest solution.